### PR TITLE
Milestone2

### PR DESCRIPTION
**Pull request by Ceejayski**
In this milestone we:
- prepared the directory structure
- implemented the App component
- implemented the BooksList component
- implemented the Book component
- Implemented the BooksForm component
- defined the Redux actions and reducers
- initialized the Redux store
- rendered the component wrapped in and pass the store down to the components.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+
+const store = createStore(combineReducers, state);
 
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Provider store={store}>
+      <App />
+    </Provider>
   </React.StrictMode>,
```
Comment (george-shammar) [inline src/index.js:37]: - [ ] According to the project's instructions, this should wrap `App` in` Provider` and assign the whole thing into a variable. e.g:

```javascript
const wrappedApp = (
  <Provider store={store}>
    <App />
  </Provider>
);
```

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi @Ceejayski , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi @Ceejayski , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/index.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
-import App from './App';
+import App from './components/App';
 
 ReactDOM.render(
   <React.StrictMode>
```
[the remark this answers is not in the document]
Comment (taiwo2) [reply src/index.js:7]: @Ceejayski, please refer to the comments from the last review, let's assign the App wrapped Provider to a variable. 🙏 
```
const wrappedApp = (
  <Provider store={store}>
    <App />
  </Provider>
);
```

**Review by taiwo2 — CHANGES_REQUESTED** [review]
Hi @Ceejayski, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ceejayski) [reply src/index.js:7]: Please Look thoroughly through the App.js, you will see App wrapped and assigned clearly to a variable, you are slowing down my progress all for an unimportant feature, this is so unprofessional of you due to the fact that you reviewed without checking for changes to the file

**Review by eliasecasta — APPROVED** [review]
## Status:  Approved 🟢

Hi, @Ceejayski, 

You are actually wrapping it accordingly to the project requirements, but it is wrapped in the wrong place, in my opinion, the result is the same and that is why I will approve this project, just take in mind that the wrapping should happen in the index.js file and not in the app.js one inside the components folder according to the project requirements provided by Microverse. I am not aware of any difference or problem that this may cause, lets see if any of the past CRs know anything more about the why of this requirement @taiwo2 @george-shammar.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**